Unity-Technologies/Graphics
# [SG2] Enable node inspector, add inspector-only fields

### Purpose of this PR

This PR enables the Node Inspector overlay and allows static ports to be displayed in the inspector using the `.InspectorOnly` UI hint.

![image](https://user-images.githubusercontent.com/10332426/160453406-904b2021-6311-41aa-94a7-626f4afba8b5.png)
![image](https://user-images.githubusercontent.com/10332426/160462450-31e6e4f8-4395-4ef0-813f-fc04f125a0c4.png)
![image](https://user-images.githubusercontent.com/10332426/160462495-f98a9474-911b-4615-9ab2-165e055929c6.png)


---
### Testing status

- Node inspector is displayed when opening a shader graph - C1006012
- Node inspector follows selection focus - C1006013
  - Click on graph background, observe that title changes to "Graph Inspector"
  - Click on a node, observe that title changes to node's name
- Node inspector displays selected node's `InspectorOnly` static fields - C1006015
  - Add and select a node with no `InspectorOnly` properties such as "Sine"
    - Observe that inspector title changes and nothing shows up in the node inspector
  - Add and select a node with `InspectorOnly` properties such as "Test UI Properties"
    - Observe that only the `InspectorOnly` fields are displayed in the node inspector (for "Test UI Properties," this is "StaticInspectorIn")
    - Change the inspector field's value and observe that the preview changes accordingly (for "Test UI Properties," setting "StaticInspectorIn" to 1 should make the preview white)


---
### Comments to reviewers

A one-line fix for "Add Redirect Node" appearing in the context menu of other graph tools' edges is included. You may need to restart the editor for this to take effect.

Some important notes:

- Inspector fields have limited support for UI hints (the same as inline constant fields)
- A bigger task to unify fields using `IConstant`s was discovered. I explored it a bit, but it quickly fell out of this specific PR's scope. My findings are in the triage doc for later discussion
- Two more default sections, "Node Properties" and "Advanced Settings," have default implementations, but are disabled for now. Node Properties shows inline constant editors, and Advanced Settings shows serialized fields on our node model
